railties/lib/rails/generators/rails/app/templates/bin/setup.tt

```diff
@@ -38,6 +38,8 @@ FileUtils.chdir APP_ROOT do
   puts "\n== Removing old logs and tempfiles =="
   system! "bin/rails log:clear tmp:clear"
 
-  puts "\n== Starting development server =="
-  exec "bin/dev"
+  command = ARGV.empty? ? ["bin/dev"] : ARGV
+  puts "\n== Starting #{command.join(" ")} =="
+  STDOUT.flush # flush the output before exec(2) so that it displays
+  exec(*command)
```

Comment:
I'd prefer something more explicit here and less generic. Maybe check for `--skip-server` in ARGV and then just don't execute this block. Don't like the idea of setup executing a generic command passed in.